hoot-core/src/main/cpp/hoot/core/io/OsmJsonReader.h

```diff
@@ -0,0 +1,225 @@
+#include <QRegExp>
+#include <QString>
+#include <QStringList>
+#include <QVariant>
+
+// Standard
+#include <set>
+#include <string>
+
+// Boost
+#include <boost/property_tree/ptree.hpp>
+
+// Hoot
+#include "hoot/core/OsmMap.h"
+
+namespace hoot
+{
+
+/**
+ * This class is intended to create an OsmMap from a given json string. JSON
+ * output from the overpass-api was used as the model for development
+ * (http://overpass-api.de/output_formats.html#json).
+ *
+ * The input string must be well-formed JSON, with the exception that it can
+ * be coded using single quotes, rather than double quotes... which makes
+ * things a lot cleaner if you are hand-jamming the JSON string into c++ code.
+ * If you are using single quotes, you may escape apostrophes with a backslash.
+ * Consider this example:
+ *
+ * QString testJsonStr =
+ *   "{                                      \n"
+ *   " 'version': 0.6,                       \n"
+ *   " 'generator': 'Overpass API',          \n"
+ *   " 'osm3s': {                            \n"
+ *   "   'timestamp_osm_base': 'date',       \n"
+ *   "   'copyright': 'c 1999'               \n"
+ *   " },                                    \n"
+ *   " 'elements': [                         \n"
+ *   " {                                     \n"
+ *   "   'type': 'node',                     \n"
+ *   "   'id': -1,                           \n"
+ *   "   'lat': 2.0,                         \n"
+ *   "   'lon': -3.0                         \n"
+ *   " },                                    \n"
+ *   " {                                     \n"
+ *   "   'type': 'node',                     \n"
+ *   "   'id': -2,                           \n"
+ *   "   'lat': 3.0,                         \n"
+ *   "   'lon': -3.0,                        \n"
+ *   "   'timestamp': '2010-01-01T00:00:00Z',\n"
+ *   "   'version': 4,                       \n"
+ *   "   'changeset': 5,                     \n"
+ *   "   'user': 'somebody',                 \n"
+ *   "   'uid': 6                            \n"
+ *   " },                                    \n"
+ *   " {                                     \n"
+ *   "   'type': 'node',                     \n"
+ *   "   'id': -3,                           \n"
+ *   "   'lat': 4.0,                         \n"
+ *   "   'lon': -3.0,                        \n"
+ *   "   'tags': {                           \n"
+ *   "     'highway': 'bus_stop',            \n"
+ *   "     'name': 'Micah\\'s Street'        \n"
+ *   "   }                                   \n"
+ *   " }                                     \n"
+ *   "]                                      \n"
+ *   "}                                      \n";
+ *
+ * It's all-or-nothing, though for the quotes - don't mix and match
+ * singles and doubles! Also, be aware that this class doesn't do anything
+ * clever to handle large datasets - it simply keeps everything in memory.
+ * Be careful if you want to use it with large datasets.
+ */
+
+class OsmJsonReader
```

Comment:
Please have a look at OsmMapReader. If you implement that and register with the factory then this will become a reader that can be used w/ arbitrary files from the command line, etc.